calyx/src/ir/structure.rs

```diff
@@ -185,13 +185,13 @@ impl GetAttributes for Cell {
 
 impl Cell {
     /// Construct a cell
-    pub fn new(name: Id, prototype: CellType) -> Self {
+    pub fn new(name: Id, prototype: CellType, external: bool) -> Self {
```

Comment:
I think we should not change this interface because it's so pervasively used in the compiler and usually it won't make sense to create a new external cell during compilation.

Instead set this to be false by default and add a new `set_external()` method to the `Cell` interface and make it `pub(super)` so it is only available to the parser but not to the rest of the compiler toolchain.